### PR TITLE
Upgrade Camel -> version 4 for CamelComponent & CamelConnector

### PR DESCRIPTION
**Pull request by fhcarter**
This is upgrading into a staging branch.  We will hold off updating master until the assemblies match & are available.

Upgrade Camel support in component & connector to Camel 4.

We upgrade support and use Camel 4.4.3 (the current LTS version).  There were a few changes to the Camel API.  Most changes, though, were due to environmental things.

This version of Camel requires JDK 17, so that upgrade went into place (thanks @rblumer4 for the Jenkins upgrades).  This version also upgrades to JUnit 5, so there are myriad changes of the form 

```
assertSomething(message, expectedSomething, actualSomething) -->
assertSomething(expectedSomething, actualSomething, message)
```

as they properly put the optional parameter last.  A bunch of other -- `BeforeClass` --> `BeforeAll`, etc.

Lots of volume of change, mostly mechanical.

### CODE REVIEW

File: camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelDiscovery.java [hunk 1]
```diff
@@ -72,25 +77,28 @@ Map<String, Set<String>> performComponentDiscovery(RouteBuilder rb, Properties p
         DefaultCamelContext ctx = new DefaultCamelContext();
     
         try  {
-            ExtendedCamelContext ectx = ctx.adapt(ExtendedCamelContext.class);
-            EnumeratingComponentResolver ecr = new EnumeratingComponentResolver(ectx.getComponentResolver());
-            ectx.setComponentResolver(ecr);
-            EnumeratingDataFormatResolver edfr = new EnumeratingDataFormatResolver(ectx.getDataFormatResolver());
-            ectx.setDataFormatResolver(edfr);
+            ExtendedCamelContext ectx = ctx.getCamelContextExtension(); //ExtendedCamelContext.class);
```
Comment (fhcarter) [inline camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelDiscovery.java]: Api change for how you get extended contexts.

File: camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelDiscovery.java [hunk 2]
```diff
@@ -72,25 +77,28 @@ Map<String, Set<String>> performComponentDiscovery(RouteBuilder rb, Properties p
...
+            
+            EnumeratingComponentResolver ecr =
+                    new EnumeratingComponentResolver(PluginHelper.getComponentResolver(ectx));
+            ectx.addContextPlugin(ComponentResolver.class, ecr);
+            EnumeratingDataFormatResolver edfr =
+                    new EnumeratingDataFormatResolver(PluginHelper.getDataFormatResolver(ectx));
+            ectx.addContextPlugin(DataFormatResolver.class, edfr);
             EnumeratingConfigurerResolver epcr = new EnumeratingConfigurerResolver();
-            ectx.setConfigurerResolver(epcr);
+            ectx.addContextPlugin(ConfigurerResolver.class, epcr);
```
Comment (fhcarter) [inline camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelDiscovery.java:88]: API change for how you update internal components

File: camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelResolver.java [hunk 3]
```diff
@@ -243,7 +248,8 @@ Collection<File> resolve(String organization, String name, String revision, Stri
                 // We should get called only with the correct set, but I have seen things behave differently, so
                 // better to be safe.  So here we check that the passed-in dependency (target*) match the type
                 // overrides for which this DependencyDescriptorMediator is being constructed.
-                if (parts[0].equals(targetOrg) && parts[1].equals(targetName) && parts[2].equals(targetRev)) {
```
Comment (fhcarter) [inline camelConnector/src/main/java/io/vantiq/extsrc/camelconn/discover/CamelResolver.java:246]: Allow for version-independent specification here.

File: camelConnector/src/main/resources/artifactMap.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-    "camelVersion": "3.21.0",
+    "camelVersion": "4.4.3",
```
Comment (fhcarter) [inline camelConnector/src/main/resources/artifactMap.json:2]: Lots of component changes due to version changes.  We discovery these via camel code, so this just updates to facts.

**Comment by fhcarter** [comment]
[Jenkins run](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/257/)

File: camelConnector/src/main/resources/log4j2.properties [hunk 5]
```diff
@@ -10,6 +10,10 @@ logger.vantiqcamelconn.name = io.vantiq.extsrc.camelconn
 logger.vantiqcamelconn.level = DEBUG
 logger.vantiqcamelconn.appenderRef = out
 
+#logger.vantiqcameldiscovery.name = io.vantiq.extsrc.camelconn.discover
+#logger.vantiqcameldiscovery.level = TRACE
+#logger.vantiqcameldiscovery.appenderRef = out
+
```
Comment (AlexBlumer) [inline camelConnector/src/main/resources/log4j2.properties:16]: Should this be removed, or were you planning on keeping it for later?

**Review by AlexBlumer — APPROVED** [review]
LGTM

**Pull request merged**